ReactWindows/ReactNative/Views/Modal/ReactModalShadowNode.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Windows.UI.ViewManagement;
+using Facebook.Yoga;
+using ReactNative.UIManager;
+
+namespace ReactNative.Views.Modal
+{
+    /// <summary>
+    /// The shadow node implementation for Picker views.
+    /// </summary>
+    public class ReactModalShadowNode : LayoutShadowNode
+    {
+        /// <summary>
+        /// Instantiates the <see cref="ReactModalShadowNode"/>.
+        /// </summary>
+        public ReactModalShadowNode()
+            : base(false, true)
+        {
+            ApplicationView.GetForCurrentView().VisibleBoundsChanged += ApplicationViewOnVisibleBoundsChanged;
```

Comment:
>ApplicationView.GetForCurrentView().VisibleBoundsChanged [](start = 12, length = 56)

This won't work, unfortunately. This code is executed by layout manager thread.. It is a "Dispatcher" thread but with no active window.
You may have fallen into the trap of the DEBUG build that uses same main UI thread as LM thread as a workaround for some OS/VS/Debugging issue (Release has the CREATE_LAYOUT_THREAD defined)

Reply:
I just tested this running the Visual Studio release configuration, which I can see is enabling the CREATE_LAYOUT_THREAD, but the VisibleBoundsChanged event is still firing and resizing the dialog, am I missing something?

Reply:
After some more testing on a phone I noticed a few sizing issues. While fixing these issues I found a way to remove the VisibleBoundsChanged event.

Reply:
@howlettt I tested your branch since you made me curious about your findings. Even though you removed the bounds change, you still have the "ApplicationView.GetForCurrentView().VisibleBounds" when adding the child.
It's true that he visible bounds of the LM view seem to mimic the main view, looks like a code hazard to me.
The problem becomes more visible when you open new views. For ex, in RNTester:
- Go to last item in list and choose "Multi Window", and then "New top level window". That creates a clone like RNTester window.
- Size the 2 windows differently.
- You'll see that LMView.VisibleBounds returns the values for main view even if you actually present the modal from the second.

This is not correct conceptually, though to be honest I played with various knobs and things look nice on screen in all cases (you may know better than me when those dimensions return by the view really matter).

For RootViews RN cheats since the sizes come from JS side and UIImplementation just shoves them into the shadow nodes.

Reply:
Hmm that could explain a oddity I saw (and had to work around) with the VisibleBounds returning slight incorrect sizing. The VisibleBounds call in the shadow node is currently only required for phone, desktop works fine without it. I haven't been able to find any other way to get phone to play nice. This solution is based off the Android code [ModalHostShadowNode.java](https://github.com/facebook/react-native/blob/1151c096dab17e5d9a6ac05b61aacecd4305f3db/ReactAndroid/src/main/java/com/facebook/react/views/modal/ModalHostShadowNode.java) where they do a similar sizing call. I'll play with it some more to see if I get an idea of how to remove this VisibleBounds call, but currently I don't have any theories.